ci.py

```diff
@@ -264,29 +268,52 @@ def get_dirty_for_module(files, module=None):
 def run_build(modules):
     log_dir = BUILD_LOG_DIR
     build_args = ['dbuild', '-sd', '--build-log-dir', log_dir, 'build', 'all', '+', ':ci-cd'] + modules
-    LOG.info('build command:', build_args)
+    LOG.debug('Executing build command: {0}\n'.format(' '.join(build_args)))
 
-    p = subprocess.Popen(build_args)
+    p = subprocess.Popen(build_args, stdout=subprocess.PIPE, universal_newlines=True)
 
     def kill(signal, frame):
         p.kill()
         LOG.warn('Finished by Ctrl-c!')
-        sys.exit(1)
+        sys.exit(2)
 
     signal.signal(signal.SIGINT, kill)
-    if p.wait() != 0:
-        LOG.error('build failed, exiting!')
-        sys.exit(p.returncode)
+
+    start_time = datetime.datetime.now()
+    while True:
+        output = p.stdout.readline()
+        print("  " + output.strip())
+        return_code = p.poll()
+        if return_code is not None:
+            LOG.debug('Return code: {0}'.format(return_code))
+            if return_code != 0:
+                LOG.error('BUILD FAILED !!!')
+                raise TempestTestFailedException() #TODO add build exc
+            if return_code == 0:
+                LOG.info('Build succeeded')
+            # Process has finished, read rest of the output 
+            for output in p.stdout.readlines():
+                LOG.debug(output.strip())
+            break
+        end_time = start_time + datetime.timedelta(minutes=20)
+        if datetime.datetime.now() >= end_time:
```

Comment:
You can replace this line wiht 
```suggestion
        if datetime.datetime.now() - start time >= datetime.timdelta(minutes(20):
```
and remove line 298. 
I think is easier to understated this line, but both option are good.

Reply:
Then let's leave with the `end_time` var